File: 1_剑指offer/25_删除链表的节点_双指针.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteNode(ListNode head, int val) {
        if(head.next == null)  return head;
        if (head.val == val) return head.next;
        ListNode pro = head.next;
        ListNode re = head;
        while(pro != null && pro.val != val){
            re = pro;
            pro = pro.next;
        }
        if (pro!=null) re.next = pro.next;
        return head;

    }
}

File: 1_剑指offer/25_删除链表的节点_双指针.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, head: ListNode, val: int) -> ListNode:
        if head==None:
            return head
        if head.val == val:
            return head.next
        pro = head.next
        ra = head
        while(pro != None and pro.val != val):
            ra = pro
            pro = pro.next
        if pro != None:
            ra.next = pro.next
        return head